protocol/rest/rest_protocol.go

```diff
@@ -0,0 +1,150 @@
+	"sync"
+	"time"
+)
+
+import (
+	"github.com/apache/dubbo-go/common"
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/config"
+	"github.com/apache/dubbo-go/protocol"
+	"github.com/apache/dubbo-go/protocol/rest/rest_interface"
+	_ "github.com/apache/dubbo-go/protocol/rest/rest_server"
+)
+
+var (
+	restProtocol *RestProtocol
+)
+
+const REST = "rest"
+
+func init() {
+	extension.SetProtocol(REST, GetRestProtocol)
+}
+
+type RestProtocol struct {
+	protocol.BaseProtocol
+	serverLock sync.Mutex
+	serverMap  map[string]rest_interface.RestServer
+	clientLock sync.Mutex
+	clientMap  map[rest_interface.RestOptions]rest_interface.RestClient
+}
+
+func NewRestProtocol() *RestProtocol {
+	return &RestProtocol{
+		BaseProtocol: protocol.NewBaseProtocol(),
+		serverMap:    make(map[string]rest_interface.RestServer, 8),
+		clientMap:    make(map[rest_interface.RestOptions]rest_interface.RestClient, 8),
+	}
+}
+
+func (rp *RestProtocol) Export(invoker protocol.Invoker) protocol.Exporter {
+	url := invoker.GetUrl()
+	serviceKey := url.ServiceKey()
+	exporter := NewRestExporter(serviceKey, invoker, rp.ExporterMap())
+	restServiceConfig := GetRestProviderServiceConfig(strings.TrimPrefix(url.Path, "/"))
+	if restServiceConfig == nil {
+		logger.Errorf("%s service doesn't has provider config", url.Path)
+		return nil
+	}
+	rp.SetExporterMap(serviceKey, exporter)
+	restServer := rp.getServer(url, restServiceConfig.Server)
+	restServer.Deploy(invoker, restServiceConfig.RestMethodConfigsMap)
+	return exporter
+}
+
+func (rp *RestProtocol) Refer(url common.URL) protocol.Invoker {
+	// create rest_invoker
+	var requestTimeout = config.GetConsumerConfig().RequestTimeout
+	requestTimeoutStr := url.GetParam(constant.TIMEOUT_KEY, config.GetConsumerConfig().Request_Timeout)
+	connectTimeout := config.GetConsumerConfig().ConnectTimeout
+	if t, err := time.ParseDuration(requestTimeoutStr); err == nil {
+		requestTimeout = t
+	}
+	restServiceConfig := GetRestConsumerServiceConfig(strings.TrimPrefix(url.Path, "/"))
+	if restServiceConfig == nil {
+		logger.Errorf("%s service doesn't has consumer config", url.Path)
+		return nil
+	}
+	restOptions := rest_interface.RestOptions{RequestTimeout: requestTimeout, ConnectTimeout: connectTimeout}
+	restClient := rp.getClient(restOptions, restServiceConfig.Client)
+	invoker := NewRestInvoker(url, &restClient, restServiceConfig.RestMethodConfigsMap)
+	rp.SetInvokers(invoker)
+	return invoker
+}
+
+func (rp *RestProtocol) getServer(url common.URL, serverType string) rest_interface.RestServer {
+	restServer, ok := rp.serverMap[url.Location]
+	if ok {
+		return restServer
+	}
+	_, ok = rp.ExporterMap().Load(url.ServiceKey())
+	if !ok {
+		panic("[RestProtocol]" + url.ServiceKey() + "is not existing")
+	}
+	rp.serverLock.Lock()
+	defer rp.serverLock.Unlock()
+	restServer, ok = rp.serverMap[url.Location]
+	if !ok {
```

Comment:
this if condition clause is so long. pls change it as follows,

if ok {
return restServer
}

xxxxx